src/archetype.rs

```diff
@@ -274,6 +274,23 @@ impl Archetype {
     }
 }
 
+impl Archetype {
+    pub unsafe fn drop<C>(&mut self, idx: u32)
+    where
+        C: 'static,
+    {
+        if needs_drop::<C>() {
```

Comment:
This should ensure that there is no measurable overhead when components are trivial types.